framework/src/Volo.Abp.Core/Volo/Abp/ObjectHelper.cs

```diff
@@ -27,37 +26,53 @@ public static class ObjectHelper
         Func<TObject, TValue> valueFactory,
         params Type[]? ignoreAttributeTypes)
     {
-        var cacheKey = $"{obj?.GetType().FullName}-" +
-                       $"{propertySelector}-" +
-                       $"{(ignoreAttributeTypes != null ? "-" + string.Join("-", ignoreAttributeTypes.Select(x => x.FullName)) : "")}";
+        var cacheKey =
+            $"{obj?.GetType().FullName}-{propertySelector}-{(ignoreAttributeTypes != null ? "-" + string.Join("-", ignoreAttributeTypes.Select(x => x.FullName)) : "")}";
 
-        var property = CachedObjectProperties.GetOrAdd(cacheKey, () =>
+        var property = _cachedObjectProperties.GetOrAdd(cacheKey, PropertyFactory);
+
+        property?.SetValue(obj, valueFactory(obj));
+        return;
+
+        PropertyInfo? PropertyFactory(string _)
         {
-            if (propertySelector.Body.NodeType != ExpressionType.MemberAccess)
+            MemberExpression? memberExpression;
+            switch (propertySelector.Body.NodeType)
+            {
+                case ExpressionType.Convert: {
+                    memberExpression = propertySelector.Body.As<UnaryExpression>().Operand as MemberExpression;
+                    break;
+                }
```

Comment:
Can you share some test code to cover this case?

Reply:
I've integrated the test to cover the explicit case.